improver/utilities/mask.py

```diff
@@ -0,0 +1,71 @@
+# (C) Crown copyright, Met Office. All rights reserved.
+#
+# This file is part of IMPROVER and is released under a BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Module for applying mask to a cube."""
+
+from typing import Union
+
+import iris
+import numpy as np
+
+from improver.utilities.common_input_handle import as_cubelist
+from improver.utilities.cube_checker import find_dimension_coordinate_mismatch
+from improver.utilities.cube_manipulation import (
+    enforce_coordinate_ordering,
+    get_coord_names,
+)
+
+
+def apply_mask(
+    *cubes: Union[iris.cube.CubeList, iris.cube.Cube],
+    mask_name: str,
+    invert_mask: bool = False,
+) -> iris.cube.Cube:
+    """
+    Apply a provided mask to a cube. If invert_mask is True, the mask will be inverted.
+
+    Args:
+        cubes:
+            A list of iris cubes that should contain exactly two cubes: a mask and a cube
+            to apply the mask to. The cubes should have the same dimensions.
+        mask_name:
+            The name of the mask cube. It should match with exactly one of the cubes in
+            the input cubelist.
+        invert_mask:
+            If True, the mask will be inverted before it is applied.
+    Returns:
+        A cube with a mask applied to the data.
+
+    Raises:
+    ValueError: If the number of cubes provided is not equal to 2.
+
+    """
+    cubes = as_cubelist(*cubes)
+    cube_names = [cube.name() for cube in cubes]
+    if len(cubes) != 2:
+        raise ValueError(
+            f"""Two cubes are required for masking, a mask and the cube to be masked.
+                         Provided cubes are {cube_names}"""
+        )
+
+    mask = cubes.extract_cube(mask_name)
+    cubes.remove(mask)
+    cube = cubes[0]
+
+    # Ensure mask is in a boolean form and invert if requested
+    mask.data = mask.data.astype(np.bool)
```

Comment:
Should you check whether this array is a masked array, you assume that it isn't I think??

Reply:
I am sort of assuming that the mask isn't masked (it would be an odd thing to do) but if you did input a masked array I don't think it would break. The masked points would just end up with no mask is the new cube.

Reply:
Fair enough.  I typically add a masked array unittest for things myself as I find mask handling somewhat inconsistent across numpy. - up to you.